src/modules/iotjs_module_buffer.c

```diff
@@ -348,31 +364,9 @@ JS_FUNCTION(Slice) {
   int64_t start = JS_GET_ARG(1, number);
   int64_t end = JS_GET_ARG(2, number);
   size_t start_idx, end_idx;
-
-  if (start < 0) {
-    size_t len = iotjs_bufferwrap_length(buffer_wrap);
-    if ((size_t)(-start) > len) {
-      start_idx = SIZE_MAX;
-    } else {
-      start_idx = (size_t)start + len;
-    }
-  } else {
-    start_idx = (size_t)start;
-  }
-  start_idx = bound_range(start_idx, 0, iotjs_bufferwrap_length(buffer_wrap));
-
-  if (end < 0) {
-    size_t len = iotjs_bufferwrap_length(buffer_wrap);
-    if ((size_t)(-end) > len) {
-      end_idx = SIZE_MAX;
-    } else {
-      end_idx = (size_t)end + len;
-    }
-  } else {
-    end_idx = (size_t)end;
-  }
-  end_idx = bound_range(end_idx, 0, iotjs_bufferwrap_length(buffer_wrap));
-
+  size_t len = iotjs_bufferwrap_length(buffer_wrap);
+  start_idx = bufferslice_idx_setter(start, len);
+  end_idx = bufferslice_idx_setter(end, len);
```

Comment:
You can add the variable definitions to these lines, no need to have them in a separate line, for example:
```c
size_t start_idx =...;
size_t end_idx = ...;
```